### ISSUE TITLE
Add developer centric documentation for test slice annotations

### ISSUE THREAD

**Issue by philwebb** [spring-projects/spring-boot#18213]
Documentation on why there's so many annotations and how you can compose your own would be useful.

**Comment by weiwensangsang** [comment]
Do you mean that we need the document about how to use some common annotations such as **@SpringBootApplication, @EnableAutoConfiguration , @@ComponentScan, @Autowired**, so the new developer can start work quickily?

**Comment by weiwensangsang** [comment]
for example, 
**@SpringBootApplication** if you use spring boot ,this annotations  will always appear in you main function.


```
import org.springframework.boot.SpringApplication;


@SpringBootApplication
public class SearchApp implements InitializingBean {

    /**
     * Main method, used to run the application.
     *
     * @param args the command line arguments.
     */
    public static void main(String[] args) {
               //,,,
    }
}
```
and how it work, how to import in maven or gradle and so on.

**Comment by wilkinsona** [comment]
No, not really. This issue is specifically about the slice test annotations (`@DataJpaTest`, `@WebMvcTest`, etc), why each of those has as many meta-annotations as it does, and how you can go about creating your own test slice.

**Comment by weiwensangsang** [comment]
Thx for you reply. I will try to figure out this problem.

**Comment by weiwensangsang** [comment]
To start, what is "many meta-annotations" means? 
In fact, if we just use meta-annotations to configure beans, @SpringBootApplication will have much more meta-annotations than @DataJpaTest.
So I think the problem equals to **why not use senior annotations to simplify @DataJpaTest**. 

From now, @DataJpaTest has these meta-annotations.
```
@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Inherited
@BootstrapWith(DataJpaTestContextBootstrapper.class)
@ExtendWith({SpringExtension.class})
@OverrideAutoConfiguration(
    enabled = false
)
@TypeExcludeFilters({DataJpaTypeExcludeFilter.class})
@Transactional
@AutoConfigureCache
@AutoConfigureDataJpa
@AutoConfigureTestDatabase
@AutoConfigureTestEntityManager
@ImportAutoConfiguration
```

if we use the senior annotation. @DataJpaTest can be like this,

```
@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Inherited
@BootstrapWith(DataJpaTestContextBootstrapper.class)
@ExtendWith({SpringExtension.class})
@OverrideAutoConfiguration(
    enabled = false
)
@TypeExcludeFilters({DataJpaTypeExcludeFilter.class})
@AAA
```

we should add @ AAA equals to,

```
@Transactional
@AutoConfigureCache
@AutoConfigureDataJpa
@AutoConfigureTestDatabase
@AutoConfigureTestEntityManager
```

Accronding to the spring-boot reference,  slice test annotations will stop AutoConfigure at first, than add the filter so they can import test beans we want, than **import some useful beans for this test** like cache, finaly start AutoConfigure again. 
Basicly, the main job of it is to add ExcludeFilter. The problem is if we use @ AAA to simplify this annotation, **@ AAA will be only use in @DataJpaTest once** because "**import some useful beans for this test**" is differect in differect slice test annotations.

for example, @WebFluxTest , "import some useful beans for this test" will be, 

```
@AutoConfigureCache
@AutoConfigureJson
@AutoConfigureWebFlux
@AutoConfigureWebTestClient
```
It is differrent between @WebFluxTest and @DataJpaTest just because they do different test.
In conclusion, slice test annotations import differert test beans, differert test beans need differert support beans. We can simplify this just like we simplify 

```
@SpringBootConfiguration
@EnableAutoConfiguration
@ComponentScan
```

with @SpringBootApplication.

Please tell me if I work in the right way. Still have some question about this.

**Comment by wilkinsona** [comment]
That's not quite what we have in mind for this one. Given that the aim is to make some additions to the reference that explain why we, the Spring Boot team, designed part of Spring Boot in a particular way, I think it's best if one of the core team takes care of this one. Thanks anyway.
